service/middleware.go

```diff
@@ -29,6 +32,35 @@ func requestIDMiddleware() mux.MiddlewareFunc {
 	}
 }
 
+func getAllowedOrigins() []string {
+	return []string{"observableusercontent.com", "localhost"}
+}
+
+func originValidator(origin string) bool {
+	valid := false
+	unescapedOrigin, err := url.QueryUnescape(origin)
+	if err != nil {
+		return valid
+	}
+	url, err := url.Parse(unescapedOrigin)
+	if err != nil {
+		return valid
+	}
+	for _, allowedDomain := range getAllowedOrigins() {
+		if strings.HasSuffix(url.Host, allowedDomain) {
+			valid = true
+		}
+	}
+	return valid
```

Comment:
I think this would be clearer without valid.
```suggestion
	unescapedOrigin, err := url.QueryUnescape(origin)
	if err != nil {
		return false
	}
	url, err := url.Parse(unescapedOrigin)
	if err != nil {
		return false
	}
	for _, allowedDomain := range getAllowedOrigins() {
		if strings.HasSuffix(url.Host, allowedDomain) {
			return true
		}
	}
	return false
```